### PR TITLE
Migeng 412 - Move to SEDA and concurrentConsumers approach

### PR DESCRIPTION
**Pull request by jonathanvila**
https://issues.redhat.com/browse/MIGENG-412
https://issues.redhat.com/browse/MIGENG-426
Moved to SEDA approach to not suffer from the sessionTimeout on the Kafka consumer
Moved to concurrentConsumers in order to have several threads to reduce analysis time

### CODE REVIEW

File: src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java [hunk 1]
```diff
@@ -21,19 +21,27 @@
...
-            e.put("_analysisId", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString());
-            return e;
-        }).map(this::createVMWorkloadInventoryModel).collect(Collectors.toList());
+        List<VMWorkloadInventoryModel> vmWorkloadInventoryModels = vmList.stream()
+                .peek(e -> {
+                    e.put("_analysisId", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString());
+                    e.put("vmEmsCluster", readValueFromExpandedEnvVarPath(VMEMSCLUSTERPATH, e));
+                    e.put("ems_cluster_id", readValueFromExpandedEnvVarPath(EMSCLUSTERIDPATH, e));
+                })
+                .peek(e -> log.debug("------- Treating Analysis {} VM :{} from {} : ", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString(), vmList.indexOf(e), vmList.size()))
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: could this `peek` be "merged" into the previous one?
Or could the previous `peek` become a `map`?

Comment (mrizzi) [inline src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: `if (log.isDebugEnabled())` will avoid the `vmList.indexOf(e)` evaluation

**Review by mrizzi — COMMENTED** [review]
@jonathanvila i feel we already talked a lot about these changes over the last weeks so i've just reported some minor issues.

File: src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java (hunk 1 above)
Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: Peek : doesnt modify the output in the stream
Map : modifies the output in the stream

So, if you want to have a debug line, followed for a processing instruction you can not use MAP

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: regarding merging it, well it's a matter of taste. 
Here there are 2 logics that I want to have : 1 is enriching the map , and the other is to log a debug info. And I have separated to show the 2 logics.

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: I dont understand

**Review by m-brophy — APPROVED** [review]

File: src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java (hunk 1 above)
Comment (mrizzi) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: `vmList.indexOf(e)` is executed every time, no matter the log level.

Comment (mrizzi) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: If you think this logics division is well explained having 2 `peeks` let's keep them

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: +1

**Review by mrizzi — APPROVED** [review]
Thanks @jonathanvila :+1:

**Pull request merged**